### PR TITLE
fix: (platform) Defect fixes for platform link components

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/3086

#### Please provide a brief summary of this pull request.
This PR provides fixes for defects found in link component.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/link/link.component.ts [hunk 1]
```diff
@@ -1,21 +1,3 @@
-/**
```
Comment (droshev) [inline libs/platform/src/lib/components/link/link.component.ts:1]: this should be addressed in a separate PR

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-disabled-emphasized-example.component.html [hunk 2]
```diff
@@ -1,12 +1,7 @@
 <p>
     This is an Example of
-    <fdp-link
-        [href]="'#'"
-        [linkType]="'emphasized'"
-        [title]="'Extra info as tooltip text and aria-label'"
-        [disabled]="true"
-    >
-        <fd-icon [glyph]="'chain-link'" [size]="'l'"></fd-icon>
+    <fdp-link href="#" linkType="emphasized" title="Extra info as tooltip text and aria-label" [disabled]="true">
```
Comment (droshev) [inline apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-disabled-emphasized-example.component.html:3]: Can we keep the formatting?

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-emphasized-example.component.html [hunk 3]
```diff
@@ -1,11 +1,7 @@
 <p>
     This is an Example of
-    <fdp-link
-        [href]="'http://www.google.com'"
-        [title]="'Extra info as tooltip text and aria-label'"
-        [linkType]="'emphasized'"
-    >
-        <fd-icon [glyph]="'chain-link'" [size]="'l'"></fd-icon>
+    <fdp-link href="http://www.google.com" title="Extra info as tooltip text and aria-label" linkType="emphasized">
```
Comment (droshev) [inline apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-emphasized-example.component.html:3]: can we keep the formatting?

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-standard-example.component.html [hunk 4]
```diff
@@ -1,79 +1,66 @@
...
 <br />
 <br />
 <p>Example of using link to download file.</p>
-<fdp-link
-    [id]="'start-download'"
-    [href]="'assets/headshot-male.jpg'"
-    [title]="'click link to start download'"
-    [download]="'pagedownloaded.jpeg'"
->
+<fdp-link id="start-download" href="assets/headshot-male.jpg" title="click link to start download"
```
Comment (droshev) [inline apps/docs/src/app/platform/component-docs/platform-link/platform-link-examples/platform-link-standard-example.component.html:35]: can we keep the formatting

File: apps/docs/src/app/platform/component-docs/platform-menu-button/platform-menu-button-examples/platform-menu-button-compact-example.component.html [hunk 5]
```diff
@@ -1,83 +1,40 @@
-<fdp-menu-button
-    [id]="'menu-button-0'"
-    [contentDensity]="'compact'"
-    [icon]="'world'"
-    [disabled]="false"
-    [fdpMenuTriggerFor]="basicMenu"
->
+<fdp-menu-button id="menu-button-0" contentDensity="compact" icon="world" [fdpMenuTriggerFor]="basicMenu">
```
Comment (droshev) [inline apps/docs/src/app/platform/component-docs/platform-menu-button/platform-menu-button-examples/platform-menu-button-compact-example.component.html:1]: can we keep the formatting?

File: apps/docs/src/app/platform/component-docs/platform-menu-button/platform-menu-button-examples/platform-menu-button-cozy-example.component.html [hunk 6]
```diff
@@ -1,83 +1,37 @@
-<fdp-menu-button
-    [id]="'menu-button-7'"
-    [contentDensity]="'cozy'"
-    [icon]="'world'"
-    [disabled]="false"
-    [fdpMenuTriggerFor]="basicMenu"
->
+<fdp-menu-button id="menu-button-7" icon="world" [fdpMenuTriggerFor]="basicMenu">
```
Comment (droshev) [inline apps/docs/src/app/platform/component-docs/platform-menu-button/platform-menu-button-examples/platform-menu-button-cozy-example.component.html:1]: same here

File: apps/docs/src/app/platform/component-docs/platform-menu-button/platform-menu-button-examples/platform-menu-button-example.component.html [hunk 7]
```diff
@@ -1,140 +1,89 @@
 <div>Menu Button Disabled</div>
-<fdp-menu-button
-    [id]="'menu-button-14'"
-    [contentDensity]="'cozy'"
-    [icon]="'world'"
-    [disabled]="false"
-    [fdpMenuTriggerFor]="basicMenu"
->
+<fdp-menu-button id="menu-button-14" contentDensity="cozy" icon="world" [fdpMenuTriggerFor]="basicMenu">
```
Comment (droshev) [inline apps/docs/src/app/platform/component-docs/platform-menu-button/platform-menu-button-examples/platform-menu-button-example.component.html:2]: same here

**Review by droshev — CHANGES_REQUESTED** [review]
Can we keep the formatting? If we want to change the formatting we should do it in a separate PR

File: libs/platform/src/lib/components/link/link.component.ts (hunk 1 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/link/link.component.ts]: keeping only link component changes. removed formatting changes from this PR.

**Comment by DeepakSap14** [comment]
Only keeping link component changes. removed other changes. changed PR title.

**Review by KevinOkamoto — APPROVED** [review]
Looks good.

**Review by droshev — APPROVED** [review]

**Review by sKudum — APPROVED** [review]
LGTM

**Comment by puru-hk** [comment]
Tested it is working fine.
